File: app_java/src/main/java/com/example/mvvmdemo/model/CommentEntity.java


package com.example.mvvmdemo.model;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.Index;
import androidx.room.PrimaryKey;


import java.util.Date;

@Entity(tableName = "comments",
        foreignKeys = {
                @ForeignKey(entity = ProductEntity.class,
                        parentColumns = "id",
                        childColumns = "productId",
                        onDelete = ForeignKey.CASCADE)},
        indices = {@Index(value = "productId")
        })
public class CommentEntity {
    @PrimaryKey(autoGenerate = true)
    private int id;
    private int productId;
    private String text;
    private Date postedAt;


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getProductId() {
        return productId;
    }

    public void setProductId(int productId) {
        this.productId = productId;
    }


    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Date getPostedAt() {
        return postedAt;
    }

    public void setPostedAt(Date postedAt) {
        this.postedAt = postedAt;
    }

    public CommentEntity() {
    }

    @Ignore
    public CommentEntity(int id, int productId, String text, Date postedAt) {
        this.id = id;
        this.productId = productId;
        this.text = text;
        this.postedAt = postedAt;
    }
}

File: app_kotlin/src/main/java/com/example/app_kotlin/db/entity/CommentEntity.kt
package com.example.app_kotlin.db.entity

import androidx.room.*
import com.example.app_kotlin.model.Comment

import java.util.*

@Entity(tableName = "comments", foreignKeys = [ForeignKey(entity = ProductEntity::class, parentColumns = ["id"], childColumns = ["productId"], onDelete = ForeignKey.CASCADE)], indices = [Index(value = ["productId"])])
class CommentEntity : Comment {
    @PrimaryKey(autoGenerate = true)
    private var id: Int = 0
    private var productId: Int = 0
    private var text: String? = null
    private var postedAt: Date? = null

    override fun getId(): Int {
        return id
    }

    fun setId(id: Int) {
        this.id = id
    }

    override fun getProductId(): Int {
        return productId
    }

    fun setProductId(productId: Int) {
        this.productId = productId
    }

    override fun getText(): String? {
        return text
    }

    fun setText(text: String) {
        this.text = text
    }

    override fun getPostedAt(): Date? {
        return postedAt
    }

    fun setPostedAt(postedAt: Date) {
        this.postedAt = postedAt
    }

    constructor() {}

    @Ignore
    constructor(id: Int, productId: Int, text: String, postedAt: Date) {
        this.id = id
        this.productId = productId
        this.text = text
        this.postedAt = postedAt
    }
}
